r-kujawa/laravel-payment-gateway
# Feature #8 - Add payment provider command

### **What does this PR do?** :robot:

- Adds a command that:
    - Generates a migration to add the provider
    - Scaffolds the provider-specific implementation files.
- Changes the dependency injection to interfaces instead of classes.

### **How should this be tested?** :microscope:
You may run `php artisan payment:add-provider {provider}` and see what it generates.
You may also take a look at the commands options and attempt to run it switching the options around.

### **Any background context you would like to provide?** :construction:
The way the dependency injection was implemented before this PR was overkill so I made it simple here.

### **Any questions or suggestions?** :thought_balloon:
Provide any questions or suggestions you might have about this change or future changes.

### **Does this relate to any story?** :link:
Closes #8, Relates to #21
